pypeit/core/coadd.py

```diff
@@ -605,120 +607,159 @@ def solve_poly_ratio(wave, flux, ivar, flux_ref, ivar_ref, norder, mask = None,
     return ymult, (result.x, wave_min, wave_max), flux_rescale, ivar_rescale, outmask
 
 
-def interp_oned(wave_new, wave_old, flux_old, ivar_old, mask_old, sensfunc=False):
-    '''
-    Utility routine to perform 1d linear nterpolation of spectra onto a new wavelength grid
+def interp_oned(wave_new, wave_old, flux_old, ivar_old, gpm_old, sensfunc=False):
+    """
+    Interpolate a 1D spectrum onto a new wavelength grid.
+
+    Interpolation is done using `scipy.interpolate.interp1d` with ``cubic``
+    interpolation. Any wavelengths in ``wave_new`` that are beyond the range
+    of ``wave_old`` are set to ``np.nan`` and masked via the output
+    good-pixel mask.
+
+    .. warning::
+
+        Any wavelength in ``wave_old`` that is less than 1 is assumed to
+        indicate that the wavelength is invalid!
 
     Args:
-       wave_new: ndarray, (nspec_new)
-            New wavelengths that you want to interpolate onto.
-       wave_old: ndarray, (nspec_old)
-            Old wavelength grid
-       flux_old: ndarray, (nspec_old)
-            Old flux on the wave_old grid
-       ivar_old: ndarray, (nspec_old)
+        wave_new (`numpy.ndarray`_):
```

Comment:
Can you make it more clear in the docs that the shapes of the new grid and old grid need not match.

Reply:
Done.